lib/Index/Index.cpp

```diff
@@ -1601,8 +1601,15 @@ bool IndexSwiftASTWalker::reportImplicitConformance(ValueDecl *witness, ValueDec
 bool IndexSwiftASTWalker::initIndexSymbol(ValueDecl *D, SourceLoc Loc,
                                           bool IsRef, IndexSymbol &Info) {
   assert(D);
+
+  // This *should* have been checked already, but we've definitely seen for
```

Comment:
Should we assert as well? That may help shake out where this is coming from.

Reply:
I wasn't sure about this. When I was looking through the calls, almost all of them are just a check on the location before calling this method. There's a couple times where we'd avoid a small amount of work but IMO it would be better to do that and have the checks in the one location. Everything already checks the result of `initIndexSymbol` anyway.

So I *almost* did that, but thought it would be better to get this up first.

If you disagree with that then I can add this assert in, though there's a few places that need the location check before doing that (they aren't the cause of the 0:0 I've seen because they all add additional roles, but they could maybe be hit).

Reply:
Okay, I've just changed the comment for now 👍